### PR TITLE
test: fix unstable test `TestFlashbackSchema`

### LINKED ISSUE #50153: flaky test in the TestFlashbackSchema
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 3. What did you see instead (Required)

```
    testkit.go:454: 
        	Error Trace:	pkg/testkit/testkit.go:454
        	            				pkg/executor/recover_test.go:580
        	Error:      	Error message not equal:
        	            	expected: "Can't find dropped database: db_not_exists in DDL history jobs"
        	            	actual  : "Can't find dropped database 'db_not_exists' in GC safe point 2024-01-05 17:15:12.298 +0000 UTC"
        	Test:       	TestFlashbackSchema 
```

https://tiprow.hawkingrei.com/view/gs/pingcapprow/pr-logs/pull/pingcap_tidb/49161/tiprow_fast_test/1744042583610363904#1:build-log.txt%3A8266

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #50153

Problem Summary: Do not use image in `TestFlashbackSchema`, there may be some old ddl jobs causing get a wrong error in the tests.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/recover_test.go [hunk 1]
```diff
@@ -555,7 +555,7 @@ func TestFlashbackSchema(t *testing.T) {
 		require.NoError(t, failpoint.Disable("github.com/pingcap/tidb/pkg/meta/autoid/mockAutoIDChange"))
 	}()
 
-	store := testkit.CreateMockStore(t)
+	store := testkit.CreateMockStore(t, mockstore.WithStoreType(mockstore.EmbedUnistore))
```
Comment (Defined2014) [inline pkg/executor/recover_test.go:558]: Do not use image here, there may be some old ddl jobs causing get a wrong error in the tests.

File: pkg/store/driver/main_test.go [hunk 2]
```diff
@@ -45,6 +45,7 @@ func TestMain(m *testing.M) {
 		goleak.IgnoreTopFunction("github.com/lestrrat-go/httprc.runFetchWorker"),
 		goleak.IgnoreTopFunction("go.etcd.io/etcd/client/pkg/v3/logutil.(*MergeLogger).outputLoop"),
 		goleak.IgnoreTopFunction("go.opencensus.io/stats/view.(*worker).start"),
+		goleak.IgnoreTopFunction("syscall.Syscall"),
```
Comment (Defined2014) [inline pkg/store/driver/main_test.go:48]: https://tiprow.hawkingrei.com/view/gs/pingcapprow/pr-logs/pull/pingcap_tidb/49161/tiprow_fast_test/1744042583610363904#1:build-log.txt%3A1639

Comment (Defined2014) [reply pkg/store/driver/main_test.go:48]: Ref link

**Review by tiancaiamao — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**